### PR TITLE
Feature ongoing form improvements

### PR DESCRIPTION
**Pull request by alexweber**

### CODE REVIEW

File: packages/zengine-ui-react/src/util/FocusFormErrors.tsx [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  const { errors, isSubmitting, isValidating } = useFormikContext();
+
+  useEffect(() => {
+    if (isSubmitting && !isValidating) {
+      const keys = Object.keys(errors);
+      if (keys.length > 0) {
+        const selector = `[name=${keys[0]}]`;
+        const errorElement = document.querySelector(selector) as HTMLElement;
+        if (errorElement !== null) {
+          window.scrollTo({
```
Comment (tehpsalmist) [inline packages/zengine-ui-react/src/util/FocusFormErrors.tsx:14]: this is problematic for IE. Better to use css `* { scroll-behavior: smooth }` and just use the x, y arguments for `scrollTo`.

https://developer.mozilla.org/en-US/docs/Web/API/ScrollToOptions

example:
https://github.com/Wizehive/zn-plugin-reporting-dashboard/blob/bc48bdfaaee75195147ce54545f2f3a9920ab8a7/plugins/dashboard/src/utilities/index.js#L148
https://github.com/Wizehive/zn-plugin-reporting-dashboard/blob/bc48bdfaaee75195147ce54545f2f3a9920ab8a7/plugins/dashboard/src/dashboard.css#L3

File: packages/zengine-ui-react/scripts/currencies.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+const fetch = require('node-fetch');
+const fs = require('fs');
+
+const CLIENT_ID = require('./client_id').CLIENT_ID;
+const endpoint = `https://api.zenginehq.com/v1/currencies?client_id=${ CLIENT_ID }`;
```
[the remark this answers is not in the document]
Comment (alexweber) [reply packages/zengine-ui-react/scripts/currencies.js:5]: So this was actually debated a lot  and the rationale was that we really don't want to add `znContext` to the plugin SDK at this juncture and require auth/tokens, etc.  Just wanna keep it being simple and as stand-alone as possible... If you take a gander at the client_id module it's using a silly pre-created test id so the idea is we just fetch and cache these during build so they're available and up-to-date but not real-time per-se...

File: packages/zengine-ui-react/src/util/FocusFormErrors.tsx (hunk 1 above)
Comment (alexweber) [reply packages/zengine-ui-react/src/util/FocusFormErrors.tsx:14]: Good catch, I'll update this...

File: packages/zengine-ui-react/scripts/currencies.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (tehpsalmist) [reply packages/zengine-ui-react/scripts/currencies.js:5]: Ok, I'll allow it. :P

[the remark this answers is not in the document]
Comment (alexweber) [reply packages/zengine-ui-react/scripts/currencies.js:5]: Praise you, cordial Sir.  Your mercy won't go unnoticed.  May the Gods fill your pastures with blessings!

**Pull request merged**